src/main/java/org/openmrs/maven/plugins/SetupPlatform.java

```diff
@@ -0,0 +1,260 @@
+     *
+     * @parameter expression="${session}"
+     * @required
+     */
+    private MavenSession mavenSession;
+
+    /**
+     * Interactive mode param
+     *
+     * @parameter expression="${interactiveMode}" default-value="true"
+     */
+    private String interactiveMode;
+
+    /**
+     * Server id (folder name)
+     *
+     * @parameter expression="${serverId}"
+     */
+    private String serverId;
+
+    /**
+     * Platform version
+     *
+     * @parameter expression="${version} default-value="1.11.2"
+     */
+    private String version;
+
+    /**
+     * DB Driver type
+     *
+     * @parameter expression="${dbDriver}
+     */
+    private String dbDriver;
+
+    /**
+     * DB Uri
+     *
+     * @parameter expression="${dbUri}
+     */
+    private String dbUri;
+
+    /**
+     * DB User
+     *
+     * @parameter expression="${dbUser}
+     */
+    private String dbUser;
+
+    /**
+     * DB Pass
+     *
+     * @parameter expression="${dbPassword}
+     */
+    private String dbPassword;
+
+    /**
+     * Component for user prompt
+     *
+     * @component
+     */
+    private Prompter prompter;
+
+    /**
+     * The Maven BuildPluginManager component.
+     *
+     * @component
+     * @required
+     */
+    private BuildPluginManager pluginManager;
+
+    /**
+     * Create and setup server with following parameters
+     * @param serverId
+     * @param version
+     * @param dbDriver
+     * @param dbUri
+     * @param dbUser
+     * @param dbPassword
+     * @param interactiveMode
+     * @throws MojoExecutionException
+     */
+    public Server setup(String serverId,
+                      String version,
+                      String dbDriver,
+                      String dbUri,
+                      String dbUser,
+                      String dbPassword,
+                      String interactiveMode) throws MojoExecutionException {
+        // path to omrs home
+        File omrsPath = new File(System.getProperty("user.home"), SDKConstants.OPENMRS_SERVER_PATH);
+        // check if user not set serverId parameter
+        if (serverId == null) try {
+            // default server name
+            String defaultId = "server";
+            // index for search unused server name
+            int indx = 0;
+            // find unused name for server
+            while (new File(omrsPath, defaultId).exists()) {
+                indx++;
+                defaultId = "server" + String.valueOf(indx);
+            }
+            // prompt this param
+            serverId = prompter.prompt("Define value for property 'serverId': (default: '" + defaultId + "')");
+            // check if user pressed 'enter' (default value)
+            if (serverId.equals("")) serverId = defaultId;
+        } catch (PrompterException e) {
+            e.printStackTrace();
+        }
+        // path to server with serverId
+        File serverPath = new File(omrsPath, serverId);
+        // check existence
```

Comment:
You use too many inline comments in your code. They are distracting and it takes time to read them. Please do not comment on lines that are obvious like here. I can say from Java code that it checks for existence :)